Classes/BagBuilder/AbstractBagBuilder.php

```diff
@@ -134,9 +133,14 @@ public function setSourcePages(string $sourcePagePids, int $recursionDepth = 0)
             return;
         }
 
-        $recursivePidList = DatabaseService::createRecursivePageList($sourcePagePids, $recursionDepth);
+        // Transform the incoming comma separated uids into an array
+        // Todo: Remove this after we pass an array from outside
+        $uids = GeneralUtility::intExplode(',', $sourcePagePids);
 
-        $this->whereClauseParts['pages'] = $this->tableName . '.pid IN (' . $recursivePidList . ')';
+        $recursivePidList =
+            GeneralUtility::makeInstance(PageRepository::class)->findWithinParentPages($uids, $recursionDepth);
```

Comment:
You'll need to use the oelib `PageRepository` here, not the Core's.

Reply:
Already changed it but did not included into the commit.